### PR TITLE
Add script to predict v2->v3 upgrade compatibility

### PR DESCRIPTION
**Pull request by geekzter**
Script that traverses agent pools and creates a list of agents with .NET 6 compatibility concerns

### CODE REVIEW

File: tools/FindAgentsNotCompatibleWithAgent/QueryAgentPoolsForCompatibleOS.ps1 [hunk 1]
```diff
@@ -0,0 +1,610 @@
+#!/usr/bin/env pwsh
+<# 
+.SYNOPSIS 
+    Predict whether agents will be able to upgrade frpom pipeline agent v2 to agent v3
```
Comment (merlynomsft) [inline tools/FindAgentsNotCompatibleWithAgent/QueryAgentPoolsForCompatibleOS.ps1:4]: [suggestion/not blocking] Still typo'd :)

**Review by merlynomsft — APPROVED** [review]
[suggestion / not-blocking]

In case you're looking for actual feedback / suggestions, I wrote up some powershell robustness practices on the internal azdo wiki (search for "Powershell and cmd robustness practices")

**Comment by max-zaytsev** [comment]
@geekzter thanks for doing this! 

The folder name was supposed to be something like "FindAgentsNotCompatibleWithAgent**V3**"?

**Review by max-zaytsev — APPROVED** [review]

**Review by kirill-ivlev — APPROVED** [review]

**Review by vmapetr — APPROVED** [review]

**Comment by geekzter** [comment]
@max-zaytsev That is what I had at first, but I removed it as we will have a similar need for a script when the next LTS version of .NET ships in 2 years. We can update the script then, and the old link will still work.

**Pull request merged**